src/WebJobs.Script.Scaling/AppServiceScaleHandler.cs

```diff
@@ -0,0 +1,306 @@
+                    {
+                        return false;
+                    }
+
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// This requests FE to remove this specific worker.
+        /// - FE should return success regardless if worker belongs.
+        /// - Any unexpected error will throw.
+        /// </summary>
+        public async Task RemoveWorker(string activityId, IWorkerInfo worker)
+        {
+            var stampHostName = GetStampHostName(worker.StampName);
+            var details = string.Format("remove worker request from {0}:{1}", AppServiceSettings.CurrentStampName, AppServiceSettings.WorkerName);
+            var pathAndQuery = string.Format("https://{0}/operations/removeworker/{1}/{2}?token={3}",
+                stampHostName,
+                AppServiceSettings.SiteName,
+                worker.WorkerName,
+                GetToken());
+
+            using (var response = await SendAsync(activityId, HttpMethod.Delete, pathAndQuery, worker, details))
+            {
+                response.EnsureSuccessStatusCode();
+            }
+        }
+
+        public static string GetStampHostName(string stampName)
+        {
+            return string.Format("{0}.cloudapp.net", stampName);
+        }
+
+        public static string GetToken()
+        {
+            if (string.IsNullOrEmpty(_token) || _tokenExpiredUtc < DateTime.UtcNow)
+            {
+                var expiredUtc = DateTime.UtcNow.Add(TokenValidity);
+                var token = ScaleUtils.GetToken(expiredUtc);
+                _token = WebUtility.UrlEncode(token);
+                _tokenExpiredUtc = expiredUtc;
+            }
+
+            return _token;
+        }
+
+        private static async Task<HttpResponseMessage> SendAsync(string activityId, HttpMethod method, string pathAndQuery, IWorkerInfo workerInfo, string details)
+        {
+            var client = GetHttpClient(activityId, workerInfo, details);
+            return await client.SendAsync(new HttpRequestMessage(method, pathAndQuery));
+        }
+
+        [SuppressMessage("Microsoft.Reliability", "CA2000:Dispose objects before losing scope", Justification = "By design")]
+        private static HttpClient GetHttpClient(string activityId, IWorkerInfo workerInfo, string details)
+        {
+            if (_httpClient == null)
+            {
+                var handler = _httpMessageHandler;
+                if (handler == null)
+                {
+                    var webHandler = new WebRequestHandler();
+                    if (!AppServiceSettings.ValidateCertificates.Value)
+                    {
+                        webHandler.ServerCertificateValidationCallback = ServerCertificateValidation;
+                    }
+
+                    handler = webHandler;
+                }
+
+                var client = new HttpClient(new HttpLoggingHandler(handler, activityId, workerInfo, details));
```

Comment:
This activity ID can't be cached/shared across requests, right? You'll be caching and reusing the same ID (the first) across all calls it seems?